enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,307 @@
+the system by using plugins. By executing `sos report` it will generate a
+detailed report with all the available information it can extract.
+
+Plugins have an enablement check or may be invoked manually when executing
+the command. A [plugin](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin)
+is a python script that we include as part of `sos`.
+
+There is a [plugin for OpenShift](https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py)
+already that almost fits the needs of this EP. MicroShift does not share the
+same resources, APIs and namespaces/names that the plugin assumes, as it was
+made for OpenShift clusters. However, much of the needs covered in the plugin
+are the same, so the contents of the plugin may be used too.
+
+A simple `sos report` will yield much of the information needed to debug a
+MicroShift deployment (the list of active plugins can be obtained by running
+`sos report -l`). However, there are some specifics that we need to add to a
+new plugin:
+1. Control plane component logs: apiserver, etcd, kubelet, etc. (these are
+retrieved by syslog, meaning they are mixed with other system logs).
+1. Yaml output for relevant manifests from API server.
+1. Logs for every container. These are already retrieved by CRIO plugin. May
+need easier indexing.
+1. MicroShift's configuration and version. This [enhancement](https://github.com/openshift/enhancements/pull/1304)
+might add additional information to gather: plugins, manifests, etc.
+
+Retrieving all this information involves using a kubernetes client. While
+MicroShift itself uses the go client extensively, the idea of introducing
+another command to produce this output and avoid installing `oc`/`kubectl` is
+not the best, since we would be adding runtime dependencies between a system
+component and MicroShift.
+
+### Workflow Description
+
+> _Disclaimer:_ Workflows described here illustrate manual processes.
+
+Assuming from now on MicroShift has already been installed in a R4E/RHEL based
+system and there has been some kind of issue.
+
+1. MicroShift administrator notices an issue/wants to get a system report.
+1. The administrator will log in to the failing host.
+1. The administrator will perform the debug report creation procedure:
+```bash
+sudo sos report
+```
+
+For the CI, the same procedure applies.
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are four main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. Extra required disk space.
+1. CEE having a different tool.
+1. Exposing sensitive data.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore we must ensure backwards compatibility with the plugin.
```

Comment:
Using the images and versions+tags would make sure that for specific version of MicroShift, you would run specific version of the sosreport.

Reply:
In the end MicroShift will reference a minimum version for sos so they are compatible. It is installed/upgraded as a dependency.